add accessibility data for trigger list, say if active

atari2600tim opened the pull request:
<!-- Keep the title short & concise so anyone non-technical can understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
For screen readers, there is not indication of if items in the trigger editor are active or not.  This change introduces that.  Also some inconsistent coloring of icons is improved.

#### Motivation for adding to Mudlet
closes #6166

#### Other info (issues closed, discussion etc)
I made some new functions to make Buttons, Scripts, and Timers use similar image selection logic as Triggers and Aliases currently use.  That fixes the inconsistent graying of scripts disabled by parent and that sort of thing.  In a future PR it might be rearranged some so that there aren't multiple spots doing the same thing, but for now I want this to be easier to review, so I made it similar to existing setup.  The Variables list is already accessible so leaving that alone.

Discussion on Discord came up with phrasing "activated", "deactivated", and adding the type of item in special cases.  For example a trigger with checkbox checked would be described as "activated" because you are looking at page dedicated to triggers, while a trigger chain with the funnel image would be "activated filter chain" because that is extra information.  Then if the trigger is under a parent that is turned off, visualized as a checked box that is dimmed, it is called "X in a deactivated group".

atari2600tim committed:
just adds the messages

SlySven reviewed (commented):
I realise that this is still a draft, but I would note that NVDA at least has an option to NOT read out the "AccessibleDescription" data, and if it does it'll (IIRC) read out the "AccessibleName" data first. As such I'd populate the latter with something like:
`{"Alias"|"Button"/"Menu"/"Toolbar"|"Key"|"Script"|"Timer"|"Trigger"|"Variable"/"Table"/"Function"} {Name} {"Item"/"Group"|"Toolbar"|"Menu"/"Module"} with ID number {ID}.`
and then follow that with a short `This {"Alias"|"Button" ...} is set as {active|inactive} {and so are all of it ancestors|but one or more of its ancestors are not active|and one or more of its ancestors are not|it has no ancestors}` as the description...

You will also have to consider things like that reported in #4665 - because the display in the editor does not update automatically - and getting it to do so manually is not obvious either (IIRC currently you have to click to a different view and then switch back)...

vadi2 commented:
Let's hold off on reviewing until it's ready, it's far too early to be influencing the design of this - it'll get messy if we do.

atari2600tim commented:
The current behavior already reads the name.  Although overriding that does sound like it could be a good possibility, thanks much.  I was making notes of behaviors and they both did the name and then the description, but I also have everything set to defaults, so actual users might indeed have the extras turned off.  I'll try out adjusting settings some.

atari2600tim committed:
improve all the strings

atari2600tim committed:
Update dlgTriggerEditor.cpp

atari2600tim committed:
Merge branch 'Mudlet:development' into timEditorAccessibility

atari2600tim committed:
Standarize UI files

atari2600tim committed:
Revert "Standarize UI files"

This reverts commit 87482f4174c2fd702a82a25ff5f5bef129af1b36.

atari2600tim committed:
Update dlgTriggerEditor.cpp

atari2600tim committed:
Standarize UI files

atari2600tim committed:
Standarize UI files

atari2600tim committed:
Update dlgTriggerEditor.cpp

atari2600tim committed:
Standarize UI files

atari2600tim committed:
change checked to activated

checked to activated
folder to activated folder

atari2600tim committed:
Standarize UI files

atari2600tim committed:
Standarize UI files

atari2600tim committed:
Standarize UI files

atari2600tim committed:
Standarize UI files

atari2600tim committed:
Revert "Standarize UI files"

This reverts commit 87482f4174c2fd702a82a25ff5f5bef129af1b36.

Revert "Revert "Standarize UI files""

This reverts commit 3af3a61a6879c5a8fe98f3009353eba05deaf2a7.

Revert "Standarize UI files"

This reverts commit 739def4d66edff362a4a181f0ac215540e288385.

Revert "Standarize UI files"

This reverts commit f7b3b55e4655cc1b26d464c70c7efa282563a7ec.

Revert "Standarize UI files"

This reverts commit dc4b8f4c0ee511c65ce95ed0a7e9dbdea3cd6778.

Revert "Standarize UI files"

This reverts commit 567b5f04e2e76e24a699bfca3a5a9c114d64b444.

Revert "Standarize UI files"

This reverts commit b2baaf62b7e575e045f1d38a414d25019619471c.

Revert "Standarize UI files"

This reverts commit eae28394fbeb39f22b920431ef3bcff1fa6e511b.

Revert "Standarize UI files"

This reverts commit 0df7438124e2991a9623e296d50bc89bd9893ed0.

atari2600tim committed:
Standarize UI files

atari2600tim committed:
Standarize UI files

atari2600tim committed:
Standarize UI files

atari2600tim committed:
hopefully have this one stick

This reverts commit 3f71b0a2cce2d22f3b17dc702827ff072bba0a18.

Revert "Standarize UI files"

This reverts commit 8e3f8ea0dc5a4ff4da85b76391d1c12208bdf373.

Revert "Standarize UI files"

This reverts commit 4013728df14e0646f1ca29cc25f04826337a7526.

atari2600tim committed:
Standarize UI files

atari2600tim committed:
Revert "Standarize UI files"

This reverts commit ea2cb4c95a21a8b25b2350035cc858fa18851c2c.

atari2600tim committed:
Standarize UI files

atari2600tim committed:
Merge branch 'Mudlet:development' into timEditorAccessibility

atari2600tim committed:
Update dlgTriggerEditor.cpp

activated timer or script in deactivated filter is now properly grayed and new phrasing
some other fixes of my own mistakes

atari2600tim committed:
Merge branch 'development' into timEditorAccessibility

atari2600tim committed:
Merge branch 'development' into timEditorAccessibility

atari2600tim committed:
Merge branch 'Mudlet:development' into timEditorAccessibility

SlySven commented:
> The Variables list already says checked or unchecked because it uses a checkbox instead of a color coded checkbox image, so I left those alone.

:information_source:  Just for the record the checkbox there is about whether the variable gets saved in the game save file (and restored on loading)...

atari2600tim committed:
initial logic for script, button, and timer colors

atari2600tim committed:
Merge branch 'Mudlet:development' into timEditorAccessibility

atari2600tim committed:
timer icons

ImageMagick: convert offsettimer-on.png -colorspace gray offset-on-grey.png

atari2600tim committed:
Update build-mudlet.yml

testing build

atari2600tim committed:
Revert "Update build-mudlet.yml"

This reverts commit 910a3c29ff02874dfecdb6ba738de113091bd15b.

atari2600tim committed:
Update dlgTriggerEditor.cpp

atari2600tim committed:
Merge branch 'Mudlet:development' into timEditorAccessibility

atari2600tim committed:
Update dlgTriggerEditor.cpp

atari2600tim committed:
Update dlgTriggerEditor.cpp

atari2600tim committed:
Update dlgTriggerEditor.cpp

atari2600tim committed:
list the strings

atari2600tim committed:
Merge branch 'Mudlet:development' into timEditorAccessibility

atari2600tim committed:
Update dlgTriggerEditor.cpp

atari2600tim committed:
Update dlgTriggerEditor.cpp

atari2600tim committed:
Merge branch 'Mudlet:development' into timEditorAccessibility

atari2600tim committed:
change how html stripped, preserve separations, move to announcer

atari2600tim committed:
Update dlgTriggerEditor.cpp

atari2600tim committed:
missed a debug msg

atari2600tim committed:
Merge branch 'development' into timEditorAccessibility

atari2600tim committed:
do the HTML parsing in future PR

atari2600tim committed:
Merge branch 'Mudlet:development' into timEditorAccessibility

atari2600tim committed:
Merge branch 'Mudlet:development' into timEditorAccessibility

atari2600tim committed:
Merge branch 'development' into timEditorAccessibility

atari2600tim committed:
move this part into separate PR

vadi2 commented on src/dlgTriggerEditor.cpp:
```diff
@@ -138,6 +138,20 @@ dlgTriggerEditor::dlgTriggerEditor(Host* pH)
                        "<p>Check the manual for <a href='http://wiki.mudlet.org/w/Manual:Contents'>more information</a>.</p>");
 
+    // Descriptions for screen readers, clarify to translators that the context of "activated" is current status and not confirmation of toggle.
+    descActive = tr("activated", "Item is currently on, short enough to be spoken");
```
Normally not a fan of centralising strings, but when there are 30+ instances of it, well :D

vadi2 reviewed (approved):
Apologies for the delay in reviewing!
